player/clipboard/clipboard-wayland.c

```diff
@@ -345,6 +345,12 @@ static bool clipboard_wayland_dispatch_events(struct clipboard_wayland_priv *wl,
         return false;
     }
 
+    if (fds[2].revents & (POLLERR | POLLHUP | POLLNVAL))
+        destroy_offer(wl->selection_offer);
+
+    if (fds[3].revents & (POLLERR | POLLHUP | POLLNVAL))
+        destroy_offer(wl->primary_selection_offer);
+
```

Comment:
should be below `if (fds[1].revents & POLLIN)`

Reply:
also it doesn't matter now but a structure like this would have prevented the bug in the first place:
```c
if (fds[2].revents & POLLIN)
        get_selection_data(wl, wl->selection_offer, false);
if (fds[2].revents) // something was handled, so destroy it
        destroy_offer(wl->selection_offer);
```

Reply:
Moved it below fds[1].

The check_fd function in wayland_common follows that structure, I'm not sure why it was changed here. https://github.com/mpv-player/mpv/blob/master/video/out/wayland_common.c#L2127-L2190